text/0000-custom-ca-certs.md

```diff
@@ -0,0 +1,148 @@
+# Meta
+[meta]: #meta
+- Name: Custom CA Certs
+- Start Date: 2020-03-30
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+In order to allow developers to use custom CA certs during build, this RFC:
+1. Suggests an interface between stacks and the platform to allow adding ca-certs to builds and for creating and extending builders with CA certs
+2. Suggest a `pack` UX for using certs during build and for creating and extending builders with CA certs
+
+**Note**: This RFC builds on some of the ideas proposed in https://github.com/buildpacks/rfcs/pull/23 to begin producing a generic interface for extending build/run images.
+
+# Motivation
+[motivation]: #motivation
+
+Use of custom CA certs is pervasive, especially for developers in enterprise environments. These developers need to fetch assets during build from corp artifact stores, publish to registries with custom certs, and make certs available during runtime for container images. Furthermore, developers might want to more granularly apply certificates, for example leaving certs out of a runtime container that were available during build. In order to account for the possible diversity of methods for adding CA certs in various stacks, the stack itself must embed this knowledge. But we should avoid the possibility of ad-hoc methods and stack forking, by coming up with a community solution.
+
+This solution should further be extensible for other ways of extending builders, thus should have a relatively generic interface. This RFC proposes using that interface for adding CA certs, but leaves the details of other forms of extension up to subsequent RFCs.
+
+# What it is
+[what-it-is]: #what-it-is
+
+### Specification
+Stack authors may include an `extend` executable to the build and/or run images of their stack:
+```
+/cnb/image/{build,run}/extend ({build,run}-extend-toml-file) (pkg-cache-dir) | cwd: /
```

Comment:
Maybe: `(pkg-cache-dir)` -> `(cache-dir)`
Might be worth removing it though, or at least explaining why it's not used in this RFC.